File: Programmers/Stack-Queue/Level2/StockPrice.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> prices) {
	vector<int> answer;

	for (int i = 0; i < prices.size(); i++) {
		int j;
		for (j = i + 1; j < prices.size(); j++) {
			if (prices[i] > prices[j])
				break;
		}
		if (j == prices.size())
			answer.push_back(j - i - 1);
		else
			answer.push_back(j - i);
	}
	return answer;
}

int main() {
	vector<int> p = { 1,2,3,2,3 };
	vector<int> a = solution(p);

	for (int t : a)
		printf("%d\n", t);
}

File: Programmers/Stack-Queue/Level2/StockPrice.py
def solution(prices):
    answer = []
    stack = [prices[0]]
    i = 1
    while stack:
        answer.append(0)
        for j in range(i, len(prices)):
            answer[i - 1] = j - i + 1
            if prices[j] < stack[-1]:
                break
        stack.pop()
        if i < len(prices):
            stack.append(prices[i])
        i += 1

    return answer


print(solution([1, 2, 3, 2, 3]))
